booster_nrpe/booster_nrpe.py

```diff
@@ -154,16 +132,16 @@ def read(self, data):
             response = struct.unpack(">2hih1024s", data)
         except struct.error as err:  # bad format...
             self.rc = 3
-            self.message = "Error : cannot read output from nrpe daemon ; datalen=%s : err=%s" % (len(data), err)
-            return (self.rc, self.message)
-
-        self.rc = response[3]
-        # the output is padded with \x00 at the end so
-        # we remove it.
-        self.message = re.sub('\x00.*$', '', response[4])
-        crc_orig = response[2]
+            self.message = ("Error : cannot unpack output ; "
+                            "datalen=%s : err=%s" % (len(data), err))
+        else:
+            self.rc = response[3]
+            # the output is padded with \x00 at the end so
+            # we remove it.
+            self.message = re.sub('\x00.*$', '', response[4])
```

Comment:
that line scares me everytime. \x00BLABLA\x00 will remove everything. Before there was a trim (or a split?) but we remove it because a weird bug.. Anyway ..